swe/ex2b_MMS.c

```diff
@@ -2305,6 +2412,45 @@ int main(int argc, char **argv) {
     PetscCall(VecView(natural, viewer));
     PetscCall(PetscViewerDestroy(&viewer));
     PetscCall(VecDestroy(&natural));
+
+    Vec analytical_vec;
+    PetscCall(VecDuplicate(X, &analytical_vec));
+    PetscScalar *analytical_ptr;
+
+    RDyMesh  *mesh  = &app->mesh;
+    RDyCells *cells = &mesh->cells;
+    PetscCall(VecGetArray(analytical_vec, &analytical_ptr));
+
+    PetscReal t = max_time;
+    for (PetscInt icell = 0; icell < mesh->num_cells_local; icell++) {
+      PetscReal xc    = cells->centroids[icell].X[1];
```

Comment:
It looks like the order of xc and yc is switched here when calculating the analytical solution.

Reply:
Great catch.

Reply:
I have fixed it now.